### PR TITLE
feat(app-platform): App platform/sentry app install endpoint

### PR DESCRIPTION
**Pull request by MeredithAnya**
* Adds `GET` and `POST` endpoint for App installations
  * Inherits from Organization Endpoint so `org:write` and `org:admin` scopes are needed to add an installation of an App on their organization

### CODE REVIEW

File: src/sentry/api/endpoints/organization_sentry_app_installations.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        )
+
+    @requires_feature('organizations:internal-catchall')
+    def post(self, request, organization):
+        serializer = OrganizationSentryAppInstallationsSerializer(request.DATA)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+
+        slug = serializer.object.get('slug')
+        install = Creator.run(
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_sentry_app_installations.py]: Are there any permissions around creating an installation? Should member or admin level users create them?

File: tests/sentry/api/endpoints/test_organization_sentry_app_installations.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        response = self.client.get(url, format='json')
+        assert response.status_code == 200
+        assert response.data == [{
+            'app': self.published_app.slug,
+            'organization': self.super_org.slug,
+            'uuid': self.installation.uuid,
+        }]
+
+    @with_feature('organizations:internal-catchall')
+    def test_users_only_sees_installs_on_their_org(self):
```
Comment (markstory) [inline tests/sentry/api/endpoints/test_organization_sentry_app_installations.py]: Are tests for the create endpoint coming next?

File: src/sentry/api/endpoints/organization_sentry_app_installations.py (hunk 1 above)
Comment (MeredithAnya) [reply src/sentry/api/endpoints/organization_sentry_app_installations.py]: I think it should just be the permissions here https://github.com/getsentry/sentry/blob/master/src/sentry/api/bases/organization.py#L18
If these installations are treated the same as integrations then that seems appropriate.

File: tests/sentry/api/endpoints/test_organization_sentry_app_installations.py (hunk 2 above)
Comment (MeredithAnya) [reply tests/sentry/api/endpoints/test_organization_sentry_app_installations.py]: Yis

File: tests/sentry/api/endpoints/test_organization_sentry_app_installations.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+        response = self.client.get(self.url, format='json')
+        assert response.status_code == 404
+
+
+class PostOrganizationSentryAppInstallationsTest(OrganizationSentryAppInstallationsTest):
+    @with_feature('organizations:internal-catchall')
+    def test_install_unpublished_app(self):
+        self.login_as(user=self.user)
+        app = SentryAppCreator.run(
```
Comment (markstory) [inline tests/sentry/api/endpoints/test_organization_sentry_app_installations.py]: Can anyone install an unpublished app or only the author/owner?

Comment (MeredithAnya) [reply tests/sentry/api/endpoints/test_organization_sentry_app_installations.py]: owners and managers pretty sure - added a test for members not being able to

**Review by markstory — APPROVED** [review]
Looks good to me.

File: src/sentry/api/endpoints/organization_sentry_app_installations.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+from sentry.api.serializers import serialize
+from sentry.features.helpers import requires_feature
+from sentry.mediators.sentry_app_installations import Creator
+from sentry.models import SentryAppInstallation
+
+
+class OrganizationSentryAppInstallationsSerializer(serializers.Serializer):
+    slug = serializers.RegexField(
+        r'^[a-z0-9_\-]+$',
+        max_length=50,
```
Comment (adhiraj) [inline src/sentry/api/endpoints/organization_sentry_app_installations.py]: do we want a min length on the slug?

File: src/sentry/api/endpoints/organization_sentry_app_installations.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+        return self.paginate(
+            request=request,
+            queryset=queryset,
+            order_by='-date_added',
+            paginator_cls=OffsetPaginator,
+            on_results=lambda x: serialize(x, request.user),
+        )
+
+    @requires_feature('organizations:internal-catchall')
```
Comment (adhiraj) [inline src/sentry/api/endpoints/organization_sentry_app_installations.py]: when do we use internal-catchall vs superuser checks?

Comment (MeredithAnya) [reply src/sentry/api/endpoints/organization_sentry_app_installations.py]: we are putting everything behind `internal-catchall` for now - but after it's released we will still want super users to be able to see certain things i.e. all the apps created (both unpublished and published) for _admin purposes assuming

**Pull request merged**